src/async_fuse/memfs/kv_engine.rs

```diff
@@ -514,6 +849,29 @@ mod test {
         assert_eq!(key.len(), 3);
     }
 
+    #[tokio::test]
+    async fn test_lock_unlock() {
```

Comment:
If you want to test the lock and unlock function, you can do the following actions:

1. lock a key in one thread/task.
2. in another thread/task, try to lock the same key with a short timeout for the same key, it should fail (assertion).
3. unlock the key in the first thread/task.
4. in the second thread/task, lock again, it should succeed (assertion).

You may need a channel or other tools to sync steps between these two threads/tasks.

Reply:
Updated.Please review this test again.